ui/src/plugins/com.android.OrganizeNestedTracks/index.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2025 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Trace} from '../../public/trace';
+import {TrackNode} from '../../public/workspace';
+import {PerfettoPlugin} from '../../public/plugin';
+import StandardGroupsPlugin from '../dev.perfetto.StandardGroups';
+import TraceProcessorTrackPlugin from '../dev.perfetto.TraceProcessorTrack';
+
+const NESTING_SEPARATOR: string = ' -> ';
+
+/**
+ * Organizes tracks with the name containing ' -> ' into a hierarchical
+ * structure. It creates a copy of the tracks at the end of their parent.
+ *
+ * For example, tracks with names:
+ *   - Foo -> Bar -> Track1
+ *   - Foo -> Bar -> Track2
+ *   - Foo -> Track3
+ *
+ * Will be reorganized into:
+ *   - Foo
+ *      - Bar
+ *         - Track1
+ *         - Track2
+ *      - Track3
+ */
+export default class implements PerfettoPlugin {
+  static readonly id = 'com.android.OrganizeNestedTracks';
+  static readonly dependencies = [
+    StandardGroupsPlugin,
+    TraceProcessorTrackPlugin,
+  ];
+
+  async onTraceLoad(ctx: Trace): Promise<void> {
+    ctx.commands.registerCommand({
+      id: 'com.android.OrganizeNestedTracks.organize',
+      name: 'Organize nested tracks',
```

Comment:
I would prefer a slightly more specific name here, as this command is quite specific. I'm also really starting to think we should introduce command descriptions!

Reply:
After using this plugin I realized that probably action is not the best way to invoke this as you need to manually invoke it every time.

I updated this plugin to not be a 'default' plugin, and 'organize' the tracks on startup. This way as long as the plugin is enabled the tracks will be grouped automatically.

Also updated the id to 'com.android.CreateTrackHierarchyFromNames' to make it a bit more specific